mm2src/coins/utxo/utxo_block_header_storage/mod.rs

```diff
@@ -95,3 +95,216 @@ impl BlockHeaderStorageOps for BlockHeaderStorage {
         self.inner.get_block_height_by_hash(hash).await
     }
 }
+
+#[cfg(test)]
+mod block_headers_storage_tests {
+    cfg_wasm32! {
+        use super::*;
+        use super::IDBBlockHeadersStorage;
+        use chain::BlockHeaderBits;
+        use common::log::wasm_log::register_wasm_log;
+        use mm2_test_helpers::for_tests::mm_ctx_with_custom_db;
+        use wasm_bindgen_test::*;
+        use spv_validation::work::MAX_BITS_BTC;
+
+        wasm_bindgen_test_configure!(run_in_browser);
+    }
+
+    cfg_native! {
+        use super::*;
+        use spv_validation::work::MAX_BITS_BTC;
+        use chain::BlockHeaderBits;
+        use mm2_test_helpers::for_tests::mm_ctx_with_custom_db;
+
+        use crate::utxo::utxo_block_header_storage::sql_block_header_storage::block_headers_cache_table;
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    pub(crate) async fn block_headers_storage_for_test(_for_coin: &str) -> (MmArc, IDBBlockHeadersStorage, String) {
+        let ctx = mm_ctx_with_custom_db();
+        let storage = IDBBlockHeadersStorage::new(&ctx, "RICK".to_string());
+        register_wasm_log();
+
+        (ctx, storage, "".to_string())
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    pub(crate) async fn block_headers_storage_for_test(for_coin: &str) -> (MmArc, SqliteBlockHeadersStorage, String) {
+        let ctx = mm_ctx_with_custom_db();
+        let storage = SqliteBlockHeadersStorage::in_memory(for_coin.into());
+        let table = block_headers_cache_table(for_coin);
+        storage.init().await.unwrap();
+
+        let initialized = storage.is_initialized_for().await.unwrap();
+        assert!(initialized);
+
+        (ctx, storage, table)
```

Comment:
For consistency, you can remove the table name from the function returning values.
Instead, `SqliteBlockHeadersStorage::is_table_empty` could be refactored as:
```rust
pub(crate) async fn is_empty(&self) -> bool {
        let table_name = get_table_name_and_validate(&self.ticker)?;
        let sql = format!("SELECT COUNT(block_height) FROM {table_name};");
        let conn = self.conn.lock().unwrap();
        let rows_count: u32 = conn.query_row(&sql, NO_PARAMS, |row| row.get(0)).unwrap();
        rows_count == 0
    }
```

Reply:
this is ready again @sergeyboyko0791 :)